ruby_programming/testing_with_rspec/introduction_to_rspec.md

```diff
@@ -0,0 +1,236 @@
+# Introduction to RSpec
+
+## Introduction
+In the previous lesson, we established the utility of [test-driven development](https://en.wikipedia.org/wiki/Test-driven_development) (TDD) in maintaining your code and sanity. In this lesson, we'll introduce you to your new best friend, the [RSpec](http://rspec.info/) testing framework. RSpec is one of the most popular testing frameworks, having been downloaded over [100 million times](https://rubygems.org/gems/rspec), as of this writing, and having been ported for use in [Rails testing](https://rubygems.org/gems/rspec-rails).
+
+## Learning Outcomes
+*Look through these now and use them to guide your learning. By the end of this lesson, expect to:*
+
+* Know what RSpec is
+* Know how to install RSpec
+* Understand basic RSpec syntax:
+  * `describe` 
+  * `it`
+
+## Introduction to RSpec
+
+### What is RSpec, and why RSpec?
+
+At the most basic level, RSpec is a Domain Specific Language written in Ruby, or, for the rest of us, a language specialized for a particular task. In this case, the task is testing Ruby code. The `rspec` gem comes packaged with all you need to get started, including five gems: `rspec-core`, `rspec-expectations`, `rspec-mocks`, `rspec-rails`, and `rspec-support`.
+
+At this point, you may be wondering, *Why RSpec? Surely, there are other frameworks out there*, and you'd be right. There are. In fact, at one point, Ruby came bundled with [Test::Unit]((https://github.com/test-unit/test-unit)) and later [Minitest](https://github.com/seattlerb/minitest) as part of its standard library, the latter of which lives on in Rails. If you tend to be pessimistic (I'm sorry, I meant *realistic*), then the [Wrong](https://github.com/sconover/wrong) testing framework might be your cup of tea. Or perhaps you're hungry and in the mood for something more substantial, in which case a side of [Bacon](https://github.com/chneukirchen/bacon) might be what you need. At the end of the day, it doesn't matter which framework you choose as long as you cultivate your testing skills. RSpec's wider adoption, especially in the Rails community, is certainly reason enough to warrant familiarity with it, but implementing tests should be the end-all and be-all, rather than choosing a particular framework.
+
+But enough proselytizing. Strap your helmet and buckle up; we're going to jump right in.
```

Comment:
Maybe link https://www.dictionary.com/browse/proselytizing on the crazy word?

Reply:
Is rspec-rails a part of the standard rspec gem?

Reply:
@CouchofTomato it isn't. I don't see where if/where I mentioned that, but I'll look it over real quick.

Reply:
@CouchofTomato found it :star:. Any thoughts on having the gems listed as bullets, rather than a phrase?